pkg/bucketeer/cache/processor/feature_flag_processor.go

```diff
@@ -0,0 +1,231 @@
+
+	"github.com/bucketeer-io/go-server-sdk/pkg/bucketeer/api"
+	"github.com/bucketeer-io/go-server-sdk/pkg/bucketeer/cache"
+	"github.com/bucketeer-io/go-server-sdk/pkg/bucketeer/log"
+	"github.com/bucketeer-io/go-server-sdk/pkg/bucketeer/model"
+)
+
+// FeatureFlagProcessor defines the interface for processing feature flags cache.
+//
+// In the background, the Processor polls the latest feature flags from the server,
+// and updates them in the local cache.
+type FeatureFlagProcessor interface {
+	// Run the processor to poll the latest cache from the server
+	Run()
+
+	// Close tears down all Processor activities, after ensuring that all events have been delivered.
+	Close()
+}
+
+type processor struct {
+	cache             cache.Cache
+	featureFlagsCache cache.FeaturesCache
+	pollingInterval   time.Duration
+	apiClient         api.Client
+	tag               string
+	closeCh           chan struct{}
+	loggers           *log.Loggers
+}
+
+// ProcessorConfig is the config for Processor.
+type FeatureFlagProcessorConfig struct {
+	// Cache
+	Cache cache.Cache
+
+	// PollingInterval is a interval of polling feature flags from the server
+	PollingInterval time.Duration
+
+	// APIClient is the client for Bucketeer service.
+	APIClient api.Client
+
+	// Loggers is the Bucketeer SDK Loggers.
+	Loggers *log.Loggers
+
+	// Tag is the Feature Flag tag
+	// The tag is set when a Feature Flag is created, and can be retrieved from the admin console.
+	Tag string
+}
+
+const (
+	cacheTTL                   = time.Duration(0)
+	featureFlagsIDKey          = "bucketeer_feature_flags_id"
+	featureFlagsRequestedAtKey = "bucketeer_feature_flags_requested_at"
+)
+
+func NewFeatureFlagProcessor(conf *FeatureFlagProcessorConfig) FeatureFlagProcessor {
+	p := &processor{
+		cache:             conf.Cache,
+		featureFlagsCache: cache.NewFeaturesCache(conf.Cache),
+		pollingInterval:   conf.PollingInterval,
+		apiClient:         conf.APIClient,
+		tag:               conf.Tag,
+		closeCh:           make(chan struct{}),
+		loggers:           conf.Loggers,
+	}
+	return p
+}
+
+func (p *processor) Run() {
+	go p.runProcessLoop()
+}
+
+func (p *processor) Close() {
+	p.closeCh <- struct{}{}
+}
+
+func (p *processor) runProcessLoop() {
+	defer func() {
+		p.loggers.Debug("bucketeer/cache: runProcessLoop done")
+	}()
+	ticker := time.NewTicker(p.pollingInterval)
+	defer ticker.Stop()
+	// Update the cache once when starts polling
+	if err := p.updateCache(); err != nil {
+		p.loggers.Errorf("bucketeer/cache: failed to update feature flags cache. Error: %v", err)
+	}
+	for {
+		select {
+		case <-ticker.C:
+			if err := p.updateCache(); err != nil {
+				p.loggers.Errorf("bucketeer/cache: failed to update feature flags cache. Error: %v", err)
+				continue
+			}
+		case <-p.closeCh:
+			return
+		}
+	}
+}
+
+func (p *processor) updateCache() error {
+	ftsID, err := p.getFeatureFlagsID()
+	if err != nil {
+		if err != cache.ErrNotFound {
```

Comment:
Comparison with errors using equality operators fails on wrapped errors, so it's better to use `!errors.Is(err, cache.ErrNotFound)`.

Reply:
Done.
https://github.com/bucketeer-io/go-server-sdk/pull/122/commits/3ff2e66d446694be1045e7223ef1a56d43cc84cd